src/main/java/be/seeseemelk/mockbukkit/entity/EnderSignalMock.java

```diff
@@ -0,0 +1,108 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.EnderSignal;
+import org.bukkit.entity.EntityType;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+public class EnderSignalMock extends EntityMock implements EnderSignal
+{
+
+	private Location targetLocation = null;
+	private boolean surviveAfterDeath = true;
+	private ItemStack item = new ItemStack(Material.ENDER_EYE);
+	private int life = 0;
+
+	/**
+	 * Constructs a new EntityMock on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public EnderSignalMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public @NotNull Location getTargetLocation()
+	{
+		Preconditions.checkState(targetLocation != null, "Target location has not been set!");
+		return targetLocation;
+	}
+
+	@Override
+	public void setTargetLocation(@NotNull Location location)
+	{
+		this.setTargetLocation(location, true);
+	}
+
+	@Override
+	public void setTargetLocation(@NotNull Location location, boolean update)
+	{
+		Preconditions.checkArgument(location != null, "Location cannot be null");
+		Preconditions.checkArgument(this.getWorld().equals(location.getWorld()), "Cannot target EnderSignal across worlds");
+		this.targetLocation = location;
+
+		if (update)
+		{
+			//ignored
```

Comment:
Should we implement this part of the code?
- Update life back to 0
- Set the value for "survive after death" to false given a 1/5 chance

Reply:
I didn't implement this yet because we still need to set a proper policy on handling randomness

Reply:
> we still need to set a proper policy on handling randomness

I think the policy should be to not have any randomness. In this case I don't see the property `survive after death` to be affecting our mock anyhow, I therefore don't think it matters in this scenario, but... here's some examples of a policy
#### Example 1
Here's an example which should be easy to understand / use
```java
void setSurviveAfterDeathUpdateValue(boolean surviveAfterDeathUpdateValue) 
    this.surviveAfterDeathUpdateValue = surviveAfterDeathUpdateValue;
}
```
```java
if(update) {
    // other stuff
    this.surviveAfterDeath = surviveAfterDeathUpdateValue;
}
```

#### Example 2
Make an interface which can provide a generic value (we could really just use the `Supplier` interface here). In comparison to example 1 this one allows for more complicated constructions, you could for example have a cyclic behavior, but I have no clue if there's any use for that.
```java
private DeterministicValue<Boolean> surviveAfterDeathUpdateValue = DeterministicValue.of(false);

void setSurviveAfterDeathUpdateValue(DeterministicValue<Boolean> surviveAfterDeathUpdateValue) 
    this.surviveAfterDeathUpdateValue = surviveAfterDeathUpdateValue;
}
```
```java
if(update) {
    // other stuff
    this.surviveAfterDeath = surviveAfterDeathUpdateValue.fetch();
}

Reply:
I like the second one a whole lot more tbh